### ISSUE TITLE
Header doesn't remove from request until I save it

### ISSUE THREAD

**Issue by abkarim** [usebruno/bruno#2776]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

If i uncheck a header it should not be send with the request, but it sends it with the request until I save it.

Does it work like this intentionally? It's so awkward and responsible for bug and my headache  

My OS details:
```No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 12 (bookworm)
Release:	12
Codename:	bookworm```


### .bru file to reproduce the bug

_No response_

### Screenshots/Live demo link

Video 
[Screencast from 2024-08-08 00-58-44.webm](https://github.com/user-attachments/assets/c3329510-8ae5-48d3-b9b5-ae09476f97bc)

**Comment by blackr1234** [comment]
Facing similar issue with request header.

If I edit the header value without saving it, then click the button to fire the request, it uses the old value.

Only after I save the request (Ctrl + S), then click the button the fire the request, it will then use the new value.

**Comment by PAllisonVSO** [comment]
I am having the same issue as blackr1234. If I change a header value, even if I save, it does not change unless I close and reopen the request. However, if I use a variable from the environment for the header value, it works fine.

**Comment by blackr1234** [comment]
https://github.com/usebruno/bruno/pull/2651

I tested that the issue I mentioned is fixed in `1.24.0`.

Perhaps OP can also try it out.
